eda/src/autogluon/eda/analysis/transform.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from typing import List, Optional
+
+import pandas as pd
+
+from autogluon.features import AbstractFeatureGenerator, AutoMLPipelineFeatureGenerator
+from .base import AbstractAnalysis
+from ..state import AnalysisState
+from ..state import StateCheckMixin
+
+logger = logging.getLogger(__name__)
+
+
+class ApplyFeatureGenerator(AbstractAnalysis, StateCheckMixin):
+    """
+    This wrapper provides transformed features to all `children` shadowing outer datasets with the updated one after application of FeatureGenerator.
+
+    Parameters
+    ----------
+    category_to_numbers: bool, default = False
```

Comment:
Is that possible for us to infer this automatically based on wrapped analyses? Not necessary to incorporate in this PR but something worthwhile to simplify in future

Reply:
This is option because some users would prefer one thing (keep `category` type), while others the other (get category codes).